assets/src/edit-story/app/prepublish/index.js

```diff
@@ -0,0 +1,124 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+/**
+ * Internal dependencies
+ */
+import error from './error';
+import warning from './warning';
+import guidance from './guidance';
+
+/**
+ *
+ * @typedef {import ('./types').Guidance} Guidance
+ * @typedef {import('../../types').Story} Story
+ */
+
+/**
+ *
+ * @param {Story} story The full story object being checked for guidance messages.
+ * @return {Guidance[]} The array of checklist items to be rectified.
+ */
+
+export default function prepublishChecklist(story) {
```

Comment:
Looking at the provider above and this function:

Could `prepublishChecklist` perhaps work better as a `usePrePublishChecklist` hook itself?

Otherwise, using a verb for the function name, e.g. `getPrePublishErrors()` might work better, avoiding confusing it with a component name.